controllers/users.js

```diff
@@ -0,0 +1,361 @@
+ * docs somehow get out of sync this might cause confusion.
+ */
+var mapUsers = function(users, settings, facilities, admins) {
+  var filtered = _.filter(users, function(user) {
+    return user.id.indexOf(getPrefix() + ':') === 0;
+  });
+  return _.map(filtered, function(user) {
+    var setting = getDoc(user.id, settings) || {};
+    return {
+      id: user.id,
+      rev: user.doc._rev,
+      username: user.doc.name,
+      fullname: setting.fullname,
+      email: setting.email,
+      phone: setting.phone,
+      place: getDoc(user.doc.facility_id, facilities),
+      type: getType(user.doc, admins),
+      language: { code: setting.language },
+      contact: getDoc(setting.contact_id, facilities),
+      known: user.doc.known
+    };
+  });
+};
+
+var rolesMap = {
+  'national-manager': ['kujua_user', 'data_entry', 'national_admin'],
+  'district-manager': ['kujua_user', 'data_entry', 'district_admin'],
+  'facility-manager': ['kujua_user', 'data_entry'],
+  'data-entry': ['data_entry'],
+  'analytics': ['kujua_analytics'],
+  'gateway': ['kujua_gateway']
+};
+
+var getRoles = function(type) {
+  // create a new array with the type first, by convention
+  return type ? [type].concat(rolesMap[type]) : [];
+};
+
+var getSettingsUpdates = function(data) {
+  return {
+  // Redundant, already saved in users db.
+  // name: data.name,
+    fullname: data.fullname,
+    email: data.email,
+    phone: data.phone,
+    language: data.language && data.language.code,
+    known: data.known,
+    facility_id: getDocID(data.place),
+    contact_id: getDocID(data.contact),
+    type: 'user-settings',
+  };
+};
+
+var getUserUpdates = function(id, data) {
+  return {
+    // CouchDB uses name field for authentication, it should be based on the id.
+    name: id.split(':')[1],
+    password: data.password,
+    // defaults role to district-manager
+    roles: data.type ? getRoles(data.type) : getRoles('district-manager'),
+    facility_id: getDocID(data.place),
+    known: data.known,
+    type: 'user'
+  };
+};
+
+var getPrefix = function() {
+  return 'org.couchdb.user';
+};
+
+var createID = function(name) {
+  return [getPrefix(), name].join(':');
+};
+
+var deleteUser = function(id, callback) {
+  // Potential problem here where _users database update happens but medic
+  // update fails and user is in inconsistent state. There is no way to do
+  // atomic update on more than one database with CouchDB API.
+  async.parallel([
+    function(cb){
+      db._users.get(id, function(err, user) {
+        if (err) {
+          return cb(err);
+        }
+        user._deleted = true;
+        db._users.insert(user, cb);
+      });
+    },
+    function(cb){
+      db.medic.get(id, function(err, user) {
+        if (err) {
+          return cb(err);
+        }
+        user._deleted = true;
+        db.medic.insert(user, cb);
+      });
+    }
+  ], function(err) {
+    callback(err);
+  });
```

Comment:
Yeah I was on the fence about that one. You can get stuck in a place unable to fully delete a user and making a username unusable.  I would prefer two calls to delete that might fix the problem rather than making it impossible to delete a user-settings doc because the doc in _users doesn't exist.